### PR TITLE
sort: Allow percentages as values for buffer-size

### PR DESCRIPTION
**Pull request by sallto**
Fixes issue uutils#3500 by adding support for percentags of total memory for the buffer-size flag. The sysinfo crate is used to determine the total memory of a system.

I used the sysinfo crate instead of hacking together unsafe code to work with libc. This makes the changed part crossplatform-compatible and realistically safer since sysinfo is a well maintained and actively developed library.

I used the error handling from the original code. The original program will default to using all of the memory of a system (not just available), while this implementation will panic with an error.

lmk if there are weird things I did as this is my first "production" rust code. Thanks!

### CODE REVIEW

File: src/uu/sort/src/sort.rs [hunk 1]
```diff
@@ -45,6 +45,7 @@ use std::ops::Range;
 use std::path::Path;
 use std::path::PathBuf;
 use std::str::Utf8Error;
+use sysinfo::{RefreshKind, System, SystemExt};
```
Comment (sylvestre) [inline src/uu/sort/src/sort.rs:48]: Would it be possible to do the same with a crate that we already using?
AFAIK, we aren't using sysinfo currently

**Comment by sylvestre** [comment]
Thanks
could you please add an integration test here https://github.com/uutils/coreutils/blob/main/tests/by-util/test_sort.rs too?

**Comment by sallto** [comment]
Added the tests!
FYI: The nix crate understandably breaks the macos build. With the sysinfo crate this worked. I don't know if this is a priority

**Comment by sylvestre** [comment]
yeah, it is a priority to keep all jobs green, sorry :)

**Comment by sallto** [comment]
Ok I rewrote that part with platform-specific code. I had to disable some tests on 32-bit since the CI/CD Pipeline has more than 2^32 bit of total ram.

**Comment by sallto** [comment]
What are the next steps now? I restarted the CI pipeline on the branch which made the macos build pass. So only the android test job fails (which it also does on them main branch). See https://github.com/sallto/coreutils/actions/runs/3383769224

**Review by sylvestre — CHANGES_REQUESTED** [review]
needs to be rebased

**Comment by sylvestre** [comment]
@sallto ping?

**Comment by sylvestre** [comment]
no action for a while, closing.

**Pull request closed without merging**